### PR TITLE
Q5 with err handling and testing

### PR DESCRIPTION
**Pull request by Luke95G**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -93,4 +93,47 @@ describe("api/reviews", () => {
...
+describe("/api/reviews/:review_id", () => {
+  test("should respond with a 200", () => {
+    return request(app).get("/api/reviews/3").expect(200);
+  });
+  test("should respond with a review object by a review_id", () => {
+    return request(app)
+      .get("/api/reviews/3")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.review).toBeInstanceOf(Array);
```
Comment (alexdevdavis) [inline __tests__/app.test.js:105]: I think this should only return an object, not an array; only one review per review id should ever exist

File: __tests__/app.test.js [hunk 2]
```diff
@@ -93,4 +93,47 @@ describe("api/reviews", () => {
...
+        body.review.forEach((review) => {
+          expect(review).toEqual(
+            expect.objectContaining({
+              owner: expect.any(String),
```
Comment (alexdevdavis) [inline __tests__/app.test.js:109]: as such, you can get rid of the forEach, and test against the exact object values for `review_id` 3

File: __tests__/app.test.js [hunk 3]
```diff
@@ -93,4 +93,47 @@ describe("api/reviews", () => {
...
+      });
+  });
+  test("should respond with a 400 bad request when the review_id is not valid", () => {
+    return request(app)
+      .get("/api/reviews/bonjour")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("Bad request.");
+      });
+  });
```
Comment (alexdevdavis) [inline __tests__/app.test.js:138]: great to see both 404 and 400 errors handled too. Nicely done 👍

File: model.js [hunk 4]
```diff
@@ -20,4 +20,24 @@ readReviews = () => {
...
+    LEFT JOIN comments ON comments.review_id = reviews.review_id
+    WHERE reviews.review_id = $1
+    GROUP BY reviews.review_id`,
+      [review_id]
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, message: "Page not found." });
+      }
+      return rows;
```
Comment (alexdevdavis) [inline model.js:39]: it's here that you can return `rows[0] ` so you're not dealing with an array anymore (arrays are for multiple elements, but we only ever want one from this function)

**Review by alexdevdavis — APPROVED** [review]
Nicely done on this one. 
- left some pointers regarding data types, and how we needn't use an array for this one.

File: model.js (hunk 4 above)
Comment (Luke95G) [reply model.js:39]: Lovely thanks Alex:) updated my code with your feedback and will merge now, appreciate it :)

**Pull request merged**